app/concerns/permission_management.rb

```diff
@@ -28,6 +28,29 @@ def get_permissions_for_identity_type(type:, group_id: nil)
     permissions_list
   end
 
+  # Function to get revisions from CMR.  The hash returned is the target : revision_id
+  # This function is needed to allay concurrency concerns when updating system or provider
+  # permissions.  Specifically, we need the revision_ids of permissions which exist, but
+  # the group does not have.
+  def get_revisions_for_edit(type:)
+    revision_ids = {}
+    options = { 'include_full_acl' => true,
+                'identity_type' => type,
+                'page_size' => 50 }
+    options['provider'] = current_user.provider_id if type == 'provider'
+
+    revisions_response = cmr_client.get_permissions(options, token)
+    if revisions_response.success?
+      revisions_response.body['items'].each do |permission|
+        revision_ids[permission['acl']["#{type}_identity"]['target']] = permission['revision_id']
+      end
+    else
+      Rails.logger.error("Get #{type.titleize} Revisions Fetch Error: #{permissions_response.clean_inspect}")
+      flash[:error] = revisions_response.error_message
+    end
+    revision_ids
+  end
+
   def assemble_permissions_for_table(permissions:, type:, group_id:)
```

Comment:
check if revision ids can be populated within this function

Reply:
With modifications to get_permissions and both assemble functions, I was able to remove the function I added.